src/backend/cdb/cdbpartition.c

```diff
@@ -2768,15 +2768,16 @@ partition_policies_equal(GpPolicy *p, PartitionNode *pn)
 			}
 			else
 			{
-				if (p->attrs == 0)
-					/* random policy, skip */
-					;
-				if (memcmp(p->attrs, rel->rd_cdbpolicy->attrs,
```

Comment:
Or maybe you should have just moved the condition here?
```c
				if (p->attrs != NULL && memcmp(p->attrs, rel->rd_cdbpolicy->attrs,
```

Reply:
I wanted to keep it explicit so that it would be clear how random policy is processed